core/src/main/java/org/apache/druid/timeline/partition/DimensionRangeShardSpec.java

```diff
@@ -145,29 +141,81 @@ private static ShardSpecLookup createLookup(List<? extends ShardSpec> shardSpecs
     return Collections.unmodifiableList(dimensions);
   }
 
-  private Range<String> getFirstDimRange()
+  /**
+   * Check if a given domain of Strings is a singleton set containing the given value
+   * @param rangeSet Domain of Strings
+   * @param val Value of String
+   * @return rangeSet == {val}
+   */
+  private boolean isRangeSetSingletonWithVal(RangeSet<String> rangeSet, String val)
   {
-    Range<String> range;
-    if (firstDimStart == null && firstDimEnd == null) {
-      range = Range.all();
-    } else if (firstDimStart == null) {
-      range = Range.atMost(firstDimEnd);
-    } else if (firstDimEnd == null) {
-      range = Range.atLeast(firstDimStart);
-    } else {
-      range = Range.closed(firstDimStart, firstDimEnd);
+    if (val == null) {
+      return false;
     }
-    return range;
+    return rangeSet.asRanges().equals(
+        Collections.singleton(Range.singleton(val))
+    );
   }
 
+  /**
```

Comment:
I feel we could just skip this javadoc as it is an overridden method anyway.
The class level javadoc would probably be a better place to talk about the pruning advantages and strategy.

When describing the strategy, try to provide an overview of the idea first and then dig into the specific cases (only if necessary).